### ISSUE TITLE
EnderIO fluid conduits not loaded propertly after updating to `GTNH-2.1.2.1`

### ISSUE THREAD

**Issue by Laiff** [GTNewHorizons/GT-New-Horizons-Modpack#9236]
### Your GTNH Discord Username

laiff#0136

### Your Pack Version

2.1.2.1

### Your Server

private

### Type of Server

Vanilla Forge

### Your Expectation

After updating all conduits present in world and work fine. 

Becuase I play on ultramine server I cant check this migration on forge, but expect successfull migration on ultramine core

### The Reality

```
[17:09:00] [Server thread/ERROR] [FML/]: A TileEntity blockConduitBundleTileEntity(crazypants.enderio.conduit.TileConduitBundle) has thrown an exception during loading, its state cannot be restored. Report this to the mod author
java.lang.RuntimeException: Could not create an instance of the conduit with name: crazypants.enderio.conduit.liquid.AdvancedEnderLiquidConduit
	at crazypants.enderio.conduit.ConduitUtil.readConduitFromNBT(ConduitUtil.java:292) ~[ConduitUtil.class:?]
	at crazypants.enderio.conduit.TileConduitBundle.readCustomNBT(TileConduitBundle.java:147) ~[TileConduitBundle.class:?]
	at com.enderio.core.common.TileEntityEnder.func_145839_a(TileEntityEnder.java:85) ~[TileEntityEnder.class:?]
	at net.minecraft.tileentity.TileEntity.func_145827_c(TileEntity.java:122) ~[aor.class:?]
	at net.minecraft.world.chunk.storage.AnvilChunkLoader.loadEntities(AnvilChunkLoader.java:525) ~[aqk.class:?]
	at net.minecraftforge.common.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:48) ~[ChunkIOProvider.class:?]
	at net.minecraftforge.common.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:15) ~[ChunkIOProvider.class:?]
	at net.minecraftforge.common.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:344) ~[AsynchronousExecutor.class:?]
	at net.minecraftforge.common.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:302) ~[AsynchronousExecutor.class:?]
	at net.minecraftforge.common.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:15) ~[ChunkIOExecutor.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:168) ~[ms.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.func_73158_c(ChunkProviderServer.java:142) ~[ms.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.func_73154_d(ChunkProviderServer.java:240) ~[ms.class:?]
	at net.minecraft.world.World.func_72964_e(World.java:448) ~[ahb.class:?]
	at net.minecraft.world.World.func_147438_o(World.java:2682) ~[ahb.class:?]
	at mods.railcraft.common.util.misc.ChunkManager.ticketsLoaded(ChunkManager.java:171) ~[ChunkManager.class:?]
	at net.minecraftforge.common.ForgeChunkManager.loadWorld(ForgeChunkManager.java:523) ~[ForgeChunkManager.class:?]
	at net.minecraftforge.common.ForgeInternalHandler.onDimensionLoad(ForgeInternalHandler.java:64) ~[ForgeInternalHandler.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_35_ForgeInternalHandler_onDimensionLoad_Load.invoke(.dynamic) ~[?:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) ~[ASMEventHandler.class:?]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:142) ~[EventBus.class:?]
	at org.ultramine.server.world.WorldDescriptor.initWorld(WorldDescriptor.java:239) ~[WorldDescriptor.class:?]
	at org.ultramine.server.world.WorldDescriptor.loadNow(WorldDescriptor.java:212) ~[WorldDescriptor.class:?]
	at org.ultramine.server.world.WorldDescriptor.weakLoadNow(WorldDescriptor.java:197) ~[WorldDescriptor.class:?]
	at org.ultramine.server.world.MultiWorld.handleServerWorldsInit(MultiWorld.java:150) ~[MultiWorld.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71247_a(DedicatedServer.java:588) ~[lt.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:261) ~[lt.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:400) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:719) ~[li.class:?]
Caused by: java.lang.ClassNotFoundException: crazypants.enderio.conduit.liquid.AdvancedEnderLiquidConduit
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191) ~[launchwrapper-1.11.jar:?]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_301]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_301]
	at java.lang.Class.forName0(Native Method) ~[?:1.8.0_301]
	at java.lang.Class.forName(Unknown Source) ~[?:1.8.0_301]
	at crazypants.enderio.conduit.ConduitUtil.readConduitFromNBT(ConduitUtil.java:290) ~[ConduitUtil.class:?]
	... 28 more
Caused by: java.lang.NullPointerException

```

Server cant migrate conduits on new version

### Your Proposal

fix migration to new ender fluid conduits

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [x] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [x] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by bombcar** [comment]
What version were you upgrading *from*?

**Comment by Laiff** [comment]
from `2.1.1.4QF-2`

**Comment by Laiff** [comment]
This affect all setups where this pipes is present. Simplest one 1 piece between single block water reservoir and crop-matron without any filters changed only communication channel to blue. 

On `2.1.1.4QF-2` this works fine but after update this block is empty and waila say about it `Conduit container` and this block breaks without any drops


Currently I also try to starup my server on default forge core to check is this related to UM or not

**Comment by Laiff** [comment]
Same problem if I start migration on forge directly

**Comment by KamenriderZO** [comment]
Same issue,but updated from 2.1.2.0rc
The EIO fluid filter missed all the data,made the entire base in fluid net chaos

**Comment by D-Cysteine** [comment]
Sorry, didn't know about this code! Can you try this jar and see if it fixes the migration?
[EnderIO-1.7.10-2.3.1.24.dirty.jar.zip](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/7768143/EnderIO-1.7.10-2.3.1.24.dirty.jar.zip)

**Comment by Laiff** [comment]
@D-Cysteine Yes it works! Thank you for fix this problem

**Comment by D-Cysteine** [comment]
Thanks for verifying that the fix worked!

Re: 2.1.2.0 migration issues, I don't think that Ender IO mod was substantially changed between 2.1.2.0rc and 2.1.2.1, so it's probably an unrelated issue? This is the PR that last modified the conduit types:
 * https://github.com/GTNewHorizons/EnderIO/pull/36 (merged Nov 20)

If the version of Ender IO that shipped with 2.1.2.0rc is later than that, then it must be something else that broke it
